core/src/components/action-sheet/action-sheet.tsx

```diff
@@ -239,22 +243,21 @@ export class ActionSheet implements ComponentInterface, OverlayInterface {
   }
 
   render() {
-    const { header, htmlAttributes, overlayIndex } = this;
+    const { header, htmlAttributes, overlayIndex, headerId } = this;
     const mode = getIonMode(this);
     const allButtons = this.getButtons();
     const cancelButton = allButtons.find((b) => b.role === 'cancel');
     const buttons = allButtons.filter((b) => b.role !== 'cancel');
-    const headerID = `action-sheet-${overlayIndex}-header`;
 
     return (
       <Host
         role="dialog"
         aria-modal="true"
-        aria-labelledby={header !== undefined ? headerID : null}
+        aria-labelledby={header !== undefined ? headerId : null}
         tabindex="-1"
         {...(htmlAttributes as any)}
         style={{
-          zIndex: `${20000 + this.overlayIndex}`,
```

Comment:
Are these changes required for this fix to land? If not, can we move them to a separate PR? Same goes for the changes in alert.tsx.